lib/sles4sap_publiccloud.pm

```diff
@@ -394,10 +395,14 @@ sub stop_hana {
         # -n is about stdin redirection and it is needed by -f to work
         my $crash_ssh_opts = $self->{my_instance}->ssh_opts . ' -fn -o ServerAliveInterval=2';
 
-        $self->{my_instance}->run_ssh_command(cmd => 'sudo su -c "' . $cmd . '"',
-            # timeout 0 has a special meaning within run_ssh_command:
-            # it result in not wrapping the command within timeout command line utility
-            timeout => 0,
+        $self->{my_instance}->run_ssh_command(
+            cmd => $cmd,
+            # This timeout is to ensure the run_ssh_command is executed in a reasonable amount of time.
+            # It is not about how much time the crash is expected to take to be executed remotely,
+            # as that one is configured to be executed in background.
+            # So, in theory, run_ssh_command returns immediately, and 10 has nothing to do with the value of sleep 5 executed remotely
+            # Also consider that internally run_ssh_command is using this value for two different guard mechanism.
+            timeout => 10,
```

Comment:
this is the only functional change of this PR: was `0` and it has been changed to `10`

Reply:
Just be doing that we get a huge change in the underline implementation.

Using `0` the `run_ssh_command` was using `script_run`;

changing to `10` results `run_script_command` to use `script_output`:

Code reference: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7acac70619558f7c0595be3a863e8cc620175ba0/lib/publiccloud/instance.pm#L99

We can eventually use `rc_only`

http://openqaworker15.qa.suse.cz/tests/297854/logfile?filename=autoinst-log.txt

```
[2024-09-20T15:12:31.936643+02:00] [debug] [pid:47594] tests/sles4sap/publiccloud/hana_sr_takeover.pm:62 called sles4sap_publiccloud::stop_hana -> lib/sles4sap_publiccloud.pm:399 called publiccloud::instance::run_ssh_command -> lib/publiccloud/instance.pm:99 called testapi::script_output
```

Command type at `15:12:31` 
```
[2024-09-20T15:12:31.949660+02:00] [debug] [pid:47737] <<< consoles::serial_screen::type_string(text="timeout 10 ssh -E /var/tmp/ssh_sut.log -F none -o LogLevel=DEBUG3 .....  -fn -o ServerAliveInterval=2 \"cloudadmin\@**********\" -- 'sudo su -c \"echo b > /proc/sysrq-trigger &\"'\nEOT_oFLmv\n", json_cmd_token="ZMWVBJYq", cmd="backend_type_string")
```

Immediately after,  `script_output` sent the command to get back the exit code
```
[2024-09-20T15:12:31.963072+02:00] [debug] [pid:47737] <<< consoles::serial_screen::type_string(json_cmd_token="zcKdZPcf", cmd="backend_type_string", text="echo oFLmv; bash -oe pipefail /tmp/scriptoFLmv.sh ; echo SCRIPT_FINISHEDoFLmv-\$?-\n")
```

The exit code is available 1.5 sec later
```
[2024-09-20T15:12:33.483224+02:00] [info] [pid:47737] ::: consoles::serial_screen::read_until: Matched output from SUT in 2 loops & 1.51532949088141 seconds: SCRIPT_FINISHEDoFLmv-0-
```

Execution return from `run_ssh_command` and `stop_hana` directly calls `wait_serial` to check if the prompt is ok

```
[2024-09-20T15:12:33.485312+02:00] [debug] [pid:47594] <<< testapi::wait_serial(regexp="# ", record_output=undef, quiet=undef, no_regex=0, expect_not_found=0, timeout=90, buffer_size=undef)
```

and prompt is immediately detected

```
[2024-09-20T15:12:33.487343+02:00] [info] [pid:47737] ::: consoles::serial_screen::read_until: Matched output from SUT in 2 loops & 0.000834797974675894 seconds: #
```

In this way we already know test is doing something wrong, as calling `wait_serial`, the `stop_hana` "consume" the prompt, and we are trying to address that in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20245

Then the test calls `script_run` to execute a probing of ssh port on the SUT `nc ... 22`. The first step of the internal `script_run` implementation is again to call `wait_serial` to check for the prompt

At `15:12:33` 
```
[2024-09-20T15:12:33.490792+02:00] [debug] [pid:47594] <<< testapi::wait_serial(timeout=90, buffer_size=undef, no_regex=1, expect_not_found=0, quiet=1, record_output=undef, regexp="# ")
```

And we get, 90sec later ...
```
[2024-09-20T15:14:03.582916+02:00] [debug] [pid:47594] >>> testapi::wait_serial: # : fail
```